example.go

```diff
@@ -43,75 +45,68 @@ var (
 )
 
 func init() {
-	subscriptionID := getEnvVarOrExit("AZURE_SUBSCRIPTION_ID")
-	tenantID := getEnvVarOrExit("AZURE_TENANT_ID")
+	authorizer, err := utils.GetAuthorizer(azure.PublicCloud)
+	onErrorFail(err, "GetAuthorizer failed")
 
-	oauthConfig, err := azure.PublicCloud.OAuthConfigForTenant(tenantID)
-	onErrorFail(err, "OAuthConfigForTenant failed")
-
-	clientID := getEnvVarOrExit("AZURE_CLIENT_ID")
-	clientSecret := getEnvVarOrExit("AZURE_CLIENT_SECRET")
-	spToken, err := azure.NewServicePrincipalToken(*oauthConfig, clientID, clientSecret, azure.PublicCloud.ResourceManagerEndpoint)
-	onErrorFail(err, "NewServicePrincipalToken failed")
-
-	createClients(subscriptionID, spToken)
+	subscriptionID := utils.GetEnvVarOrExit("AZURE_SUBSCRIPTION_ID")
+	createClients(subscriptionID, authorizer)
 }
 
 func main() {
-	subnetInfo, err := createNeededResources()
-	onErrorFail(err, "createNeededResources failed")
+	subnet := createNeededResources()
 	defer groupClient.Delete(groupName, nil)
 
-	err = createVM(linuxVMname, "Canonical", "UbuntuServer", "16.04.0-LTS", subnetInfo)
-	onErrorFail(err, "createVM failed")
-
-	vmOperations(linuxVMname)
+	go createVM(linuxVMname, "Canonical", "UbuntuServer", "16.04.0-LTS", subnet)
```

Comment:
Using this syntax will fire and forget that this request ever happened. Most importantly, it may lead to execution errors later when the "vm Operations" try to happen. At the moment it probably happens to work because the program waits for the windows VM to be created.

Schedule some time with me to go over how I would make this safer, @mcardosos. :)